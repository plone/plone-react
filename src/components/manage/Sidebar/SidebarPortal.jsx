import React from 'react';
import PropTypes from 'prop-types';
import { Portal } from 'react-portal';

/**
 * Portal that wraps Sidebar components
 * @param {Array} children Sidebar content
 * @param {bool} selected Sidebar needs to know when the related tile is selected
 * @returns {string} Rendered sidebar
 */
const SidebarPortal = ({ children, selected }) => (
  <>
    {selected && (
      <Portal
        node={__CLIENT__ && document.getElementById('sidebar-properties')}
      >
        <aside
          onClick={e => {
            e.stopPropagation();
          }}
          onKeyDown={e => {
            e.stopPropagation();
          }}
        >
          {children}
        </aside>
      </Portal>
    )}
  </>
);

SidebarPortal.propTypes = {
  children: PropTypes.any,
  selected: PropTypes.bool.isRequired,
};

export default SidebarPortal;
