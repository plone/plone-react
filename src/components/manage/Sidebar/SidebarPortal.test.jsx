import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import SidebarPortal from './SidebarPortal';

test('sidebar portal is rendered when the tile is selected', () => {
  const renderer = new ShallowRenderer();
  renderer.render(
    <SidebarPortal selected={true}>
      <p>Tested!</p>
    </SidebarPortal>,
  );
  const component = renderer.getRenderOutput();
  expect(component).toMatchSnapshot();
});

test('sidebar portal is not rendered when the tile is not selected', () => {
  const renderer = new ShallowRenderer();
  renderer.render(
    <SidebarPortal selected={false}>
      <p>Tested, but you shouldn't see this in the snapshot!</p>
    </SidebarPortal>,
  );
  const component = renderer.getRenderOutput();
  expect(component).toMatchSnapshot();
});
