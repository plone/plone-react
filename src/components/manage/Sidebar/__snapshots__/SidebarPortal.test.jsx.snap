// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidebar portal is not rendered when the tile is not selected 1`] = `<Fragment />`;

exports[`sidebar portal is rendered when the tile is selected 1`] = `
<Fragment>
  <Portal
    node={null}
  >
    <aside
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <p>
        Tested!
      </p>
    </aside>
  </Portal>
</Fragment>
`;
